include/common/Driver.hpp

```diff
@@ -473,16 +471,17 @@ class Driver
 		{
 			oss.str("");
 			oss << "unit_" << std::setfill('0') << std::setw(3) << std::setprecision(0) << i;
-			pp.pushScope(oss.str());
-			if (pp.exists("WRITE_SOLUTION_LAST_UNIT"))
+			if (pp.exists(oss.str()))
 			{
-				_writeLastStateUnit = pp.getBool("WRITE_SOLUTION_LAST_UNIT");
-				_flag[i] = true;
+				pp.pushScope(oss.str());
+				if (pp.exists("WRITE_SOLUTION_LAST_UNIT"))
+				{
+					_writeLastStateUnit = pp.getBool("WRITE_SOLUTION_LAST_UNIT");
+					if (_writeLastStateUnit==true)
+						_writeLastStateUnitId.push_back(i);
```

Comment:
* `_writeLastStateUnit` is only used here. Please make it a local variable:
   ```
   const bool writeState = pp.getBool("WRITE_SOLUTION_LAST_UNIT");
   ```
* Don't do `if (something == true)`, please do `if (something)` instead.